src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -251,6 +252,46 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
     }
   };
 
+  onClickTestConnection = async () => {
+    this.setState({ isLoading: true });
+    const existingAuthType = this.props.existingDataSource.auth.type;
+
+    try {
+      const isNewCredential = !!(
+        existingAuthType === AuthType.NoAuth && this.state.auth.type !== existingAuthType
+      );
+      const formValues: DataSourceAttributes = {
+        title: this.state.title,
+        description: this.state.description,
+        endpoint: this.props.existingDataSource.endpoint,
+        auth: {
+          ...this.state.auth,
+          credentials: {
+            ...this.state.auth.credentials,
+            password: isNewCredential ? this.state.auth.credentials.password : '',
+          },
+        },
+      };
+
+      await this.props.handleTestConnection(formValues);
+
+      this.props.displayToastMessage({
+        id: 'dataSourcesManagement.editDataSource.testConnectionSuccessMsg',
+        defaultMessage:
+          'Connecting to the endpoint using the provided authentication method was successful.',
+        success: true,
+      });
+    } catch (e) {
+      this.props.displayToastMessage({
+        id: 'dataSourcesManagement.editDataSource.testConnectionFailMsg',
+        defaultMessage:
+          'Failed Connecting to the endpoint using the provided authentication method.',
```

Comment:
Rewording and case; maybe `An attempt to connect to the endpoint using the provided authentication method failed`.

However, instead of `using the provided authentication method`, I would prefer to see `with authentication` and `without authentication`.

Reply:
Wording will be finalized by @KrooshalUX